src/tls/server.c

```diff
@@ -276,8 +275,9 @@ connection_init_ws (Connection *c)
     server.ws_notls = ws;
 
   /* epoll the fd */
-  ev.data.ptr = &c->buf_ws;
-  if (epoll_ctl (server.epollfd, EPOLL_CTL_ADD, fd, &ev) < 0)
+  c->buf_ws.epoll_ev.data.ptr = &c->buf_ws;
+  c->buf_ws.epoll_ev.events = EPOLLIN;
+  if (epoll_ctl (server.epollfd, EPOLL_CTL_ADD, fd, &c->buf_ws.epoll_ev) < 0)
```

Comment:
As Connection now needs the epoll_event struct anyway, this code could/should also move into connection.c. I'm not entirely sure about the epoll_ctl yet (that may stay in server.c), need to play around with this and see how it feels.

But this is independent from making it actually work on Edge.